cdap-security/src/main/java/io/cdap/cdap/security/authorization/DefaultAccessEnforcer.java

```diff
@@ -197,6 +147,49 @@ private void doEnforce(EntityType entityType, EntityId parentId, Principal princ
     }
   }
 
+  @Override
+  public Set<? extends EntityId> isVisible(Set<? extends EntityId> entityIds, Principal principal)
+    throws AccessException {
+    if (internalAuthEnabled && principal.getFullCredential() != null && principal.getFullCredential().getType()
+      == Credential.CredentialType.INTERNAL) {
```

Comment:
same as above

Reply:
Since this is a cherry-pick, I've gone ahead and opened a new PR #13533 to address these comments.